### PR TITLE
[FEATURE] Add preprocess (repos_2bow_index in ml)

### PR DESCRIPTION
**Pull request by r0mainK**
[See PR in ml FIRST](https://github.com/src-d/ml/pull/275)

### CONTEXT

To be merged after ml PR is approved, and not else wise.

### CHANGES

- main: added pickle_index as an optional input or output path. if supplied, the pickle will be loaded from the location, or be saved - **in which case the metadata and bags will not be written**
- bags: logic of above functionality

### CODE REVIEW

**Comment by r0mainK** [comment]
OKay going to update this PR so it can get merged.

File: apollo/__main__.py [hunk 1]
```diff
@@ -96,6 +106,9 @@ def add_template_args(my_parser, default_template):
     add_cassandra_args(source2bags_parser)
     add_df_args(source2bags_parser)
     add_repartitioner_arg(source2bags_parser)
+    source2bags_parser.add_argument(
```
Comment (vmarkovtsev) [inline apollo/__main__.py]: Looks like we should move it to sourced-ml and re-use here

Comment (r0mainK) [reply apollo/__main__.py]: on it, will add to https://github.com/src-d/ml/pull/285

**Comment by vmarkovtsev** [comment]
Waits for https://github.com/src-d/ml/pull/285

**Pull request merged**